src/Infrastructure/Services/UserService.cs

```diff
@@ -421,6 +422,14 @@ public async Task<Response<string>> RespondToMfaChallengeAsync(string mfaCode, s
 
         private string CleanPhoneNumber(string phoneNumber)
         {
+            // ensure its a valid uk mobile using regex
+            var isValid = Regex.IsMatch(phoneNumber, @"^((\+44\s?|0)7([45789]\d{2}|624)\s?\d{3}\s?\d{3})$",
```

Comment:
The use of a single regex here could be problematic.
I suggest using [libphonenumber-csharp](https://github.com/caseykramer/libphonenumber-csharp) to parse, validate and format the number in E164 (that's what cognito is expecting).